libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Input/InputDialog.cs

```diff
@@ -487,15 +487,26 @@ protected virtual async Task<IActivity> OnRenderPromptAsync(DialogContext dc, In
                 msg.InputHint = InputHints.ExpectingInput;
             }
 
+            TrackGeneratorResultEvent(dc, template, msg);
+
+            return msg;
+        }
+
+        /// <summary>
+        /// Track GeneratorResultEvent telemetry event with InputDialogResultEvent context.
+        /// </summary>
+        /// <param name="dc">Current <see cref="DialogContext"/>.</param>
+        /// <param name="activityTemplate"><see cref="ITemplate{T}"/> used to create the Activity.</param>
+        /// <param name="msg">The <see cref="IMessageActivity"/> which will be sent.</param>
+        protected virtual void TrackGeneratorResultEvent(DialogContext dc, ITemplate<Activity> activityTemplate, IMessageActivity msg)
```

Comment:
Consider making this method internal so we don't add public APIs. 

```suggestion
        internal virtual void TrackGeneratorResultEvent(DialogContext dc, ITemplate<Activity> activityTemplate, IMessageActivity msg)
```